specs/commandline/sign.md

```diff
@@ -153,6 +165,30 @@ Warning: Always sign the artifact using digest(`@sha256:...`) rather than a tag(
 Successfully signed localhost:5000/net-monitor@sha256:b94d27b9934d3e08a52e52d7da7dabfac484efe37a5380ee9088f7ace2efcde9
 ```
 
+### Sign an arbitrary file located in file system
+Notation supports signing a file located in user's file system. The file's content (blob) gets signed and the generated signature is stored in the same file system.
+```shell
+# Prerequisites:
+# A default signing key is configured using CLI "notation key"
+
+# Use flag "--file" to enable signing a file
+# The generated signature is saved at the same dir as the target file with name
+# <target_file>.sig
+notation sign --file <target_file_path>
+
+# Use flag "--file" to enable signing a file
+# Use flag "--signature" to specify path where the generated signature is stored
+notation sign --file --signature <signature_path> <target_file_path>
```

Comment:
> What will happen if I form command like `notation sign --file <target_file_path> --signature <signature_path>`, will it succeed or fail?

This command will succeed.

> And should we use `--blob` instead of `--file`?

It was actually called `--blob`, but on a second thought I changed it to `--file`. The reason is that a user who wants to sign/verify a file might looking for a `--file` flag instead of `--blob`. In fact, they don't have to understand what a `blob` is in their scenario. All they need to know is they are signing/verifying a file.

Reply:
@priteshbandi Correct me if I'm wrong, you would like a flag to sign arbitrary things, by things I mean it can be a file, it can be data from stdin, etc.
But I think if we want to support data from stdin, then we do need a new flag for it instead of sharing the same flag with `file`. Here was the proposal from @shizhMSFT when we brought in `--oci-layout`: https://github.com/notaryproject/notation/pull/601#discussion_r1150810148. Basically, we are following the practice of `one type per flag` here. This design aligns with what we already have.
Given above, the `--blob` becomes a bit general. In the future, if we want to support sign/verify a descriptor, then we would introduce a `--descriptor` flag. If the digest of this descriptor belongs to a `blob`, then there's ambiguity, user might confuse on which flag they should use, `--blob` or `--descriptor`.
Coming back to the name `--file`, I think we can explicitly say that it sign/verify the `content` of a file in the flag's description of the CLI.